staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -154,13 +144,19 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	newReq, cancelFn := newRequestForProxy(location, req)
 	defer cancelFn()
 
-	if handlingInfo.proxyRoundTripper == nil {
+	roundTripper, err := transport.New(handlingInfo.transportConfig)
```

Comment:
previously, we re-used the same transport for all requests instead of building a fresh one.  Does this carry performance penalties in terms of object creation and GC?

@liggitt

Reply:
Yes, building this per request is expensive, especially when we set options (see comment on GetCert usage) that means every time `updateAPIService` is called we are guaranteed to no longer use the previously cached TLS config.